core/state.go

```diff
@@ -356,10 +364,26 @@ func (s *State) Class(classHash *felt.Felt) (*DeclaredClass, error) {
 
 // updateContractStorage applies the diff set to the Trie of the
 // contract at the given address in the given Txn context.
-func (s *State) updateContractStorage(stateTrie *trie.Trie, addr *felt.Felt, diff []StorageDiff, onChanged OnValueChanged) error {
+func (s *State) updateContractStorage(stateTrie *trie.Trie, addr *felt.Felt, diff []StorageDiff, onChanged OnValueChanged,
+	blockNumber uint64,
+) error {
 	contract, err := NewContract(addr, s.txn)
 	if err != nil {
-		return err
+		if _, ok := specialContracts[*addr]; !errors.Is(err, ErrContractNotDeployed) || !ok {
```

Comment:
nit: personally I think the code is easier to follow if the two cases are separated. Then we don't need to do a lookup in the map if `err` is not `ErrContractNotDeployed`.

```go
if !errors.Is(err, ErrContractNotDeployed) { return err }

if _, ok := specialContracts[*addr]; !ok { return err }
```

Reply:
I did have it like this initially but I thought the current once followed @omerfirmak suggestion more closely. Given we won't do a lookup, I will change ti back

Reply:
done